treeherder/etl/seta.py

```diff
@@ -0,0 +1,210 @@
+                # desktop-test-linux64/debug-jittests-3
+                testtype = job['job_type_name'].split(
+                    '{buildtype}-'.format(buildtype=separator))[-1]
+                job_identifier = job['ref_data_name']
+
+        return testtype, job_identifier
+
+    def query_jobtypes(self):
+        """Query all available jobtypes and return it as list"""
+        return self.jobtypes
+
+    def query_jobnames(self):
+        """Query all jobnames including buildtype and groupcode, then return them as list"""
+        return self.jobnames
+
+    def _get_jobnames(self, job, testtype, job_identifier):
+        signature = ''  # TH specific
+        buildplatform = job['build_system_type']
+        buildtype = job['platform_option']
+        platform = job['platform']
+        name = testtype
+        jobname = ''  # TH specific
+        job_identifier = job_identifier
+        group = _get_group(name)
+        groupcode = _get_group_code(self.tbplnames, name)
+        code = _get_code(self.tbplnames, name)
+        data = {'buildplatform': buildplatform, 'jobname': jobname, 'signature': signature,
+                'job_type_name': group, 'job_group_symbol': groupcode,
+                'name': name, 'job_type_symbol': code,
+                'ref_data_name': job_identifier,
+                'platform': platform, 'buildtype': buildtype}
+        return data
+
+
+def parse_testtype(build_system_type, job_type_name, platform_option, ref_data_name):
+    '''
+    build_system_type - buildbot or taskcluster
+    job_type_name - e.g. Mochitest (JobType.name)
+    platform_option - e.g. opt
+    ref_data_name - builder name
+    '''
+    # TODO: figure out how to ignore build, lint, etc. jobs
+    if build_system_type == 'buildbot':
+        return ref_data_name.split(' ')[-1]
+    else:
+        # e.g. "desktop-test-linux64/opt-reftest-8",
+        # taskcluster test types
+        testtype = job_type_name
+        if testtype.startswith('[funsize'):
+            return None
+
+        testtype = testtype.split('/opt-')[-1]
+        testtype = testtype.split('/debug-')[-1]
+
+        # this is plain-reftests for android
+        testtype = testtype.replace('plain-', '')
+
+        # SETA/Ouija do not care about builds, or other jobs like lint, etc.
+        # Bug 1318659 - for how mozci could help here
+        testtype = testtype.replace(' Opt', 'build')
+        testtype = testtype.replace(' Debug', 'build')
+        testtype = testtype.replace(' Dbg', 'build')
+        testtype = testtype.replace(' (opt)', 'build')
+        testtype = testtype.replace(' PGO Opt', 'build')
+        testtype = testtype.replace(' Valgrind Opt', 'build')
+        testtype = testtype.replace(' Artifact Opt', 'build')
+        testtype = testtype.replace(' (debug)', 'build')
+
+        testtype = testtype.strip(' ')
+
+        # TODO: these changes should have bugs on file to fix the names
```

Comment:
when BBB is doing bc-e10s and dt-e10s jobs, we will probably not need this replacement.